app/src/test/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModelTest.kt

```diff
@@ -46,27 +47,40 @@ class ProfileViewModelTest : ShouldSpec(
 			coVerify { authRepository.logout() }
 		}
 
-		should("call deleteAccount on AuthRepository AND show on ToastPublisher depending on result WHEN onDeleteAccountClick") {
+		should("call deleteAccount on AuthRepository AND show on ToastPublisher depending on result WHEN onDeleteAccountConfirmClick") {
 			listOf(
 				AccountDeletionResult.Success to R.string.DeleteAccountSuccessToastText,
 				AccountDeletionResult.ReAuthenticationNeeded to R.string.DeleteAccountReAuthenticationToastText,
 			).forAll { (result, textId) ->
 				coEvery { authRepository.deleteAccount() } returns result
 
-				tested.onDeleteAccountClick()
+				tested.onDeleteAccountConfirmClick()
 
+				tested.state.value shouldBe ProfileState(showDeleteAccountPopup = false)
```

Comment:
State is exposed from view model as non mutable flow, on one hand it's good because that's how it should be encapsulated, on the other hand it makes testing harder because we cannot set the initial state before testing. In case where we the initial values set in the ProfileState to be the same this test is not quite precise. I'm not sure how to handle that yet, maybe we should introduce something like uninitialized state as a default one set in view model?

Reply:
1. In general, I don't like the approach of fetching the data, initializing etc. in the constructor of view model. Would be nice to have more **explicit** control over it. Of course, it's nice and convenient, we don't have to call any method because an `init` block does the job. On the other hand we have messed up tests, with the lazy delegate, just to postpone the initialization and execution of the logic. 
Probably in long term, that would evolve to some sort of habit in our tests, without understanding and purpose of that.
The initial idea would be to emit an event from "UI" to initialize the data, e.g. `viewModel.emit(Init)` - that would be something similar to our well known `viewModel.onCreate`. 
2.  Back to `onDeleteAccountConfirmClick` test. Shouldn't we change to side effect and don't care about the state?

Reply:
Note to point 2 - I've forgot that dialog is not a destination from now 😄 That changes the point of view 😄

Reply:
Regarding the `init` and separate `onCreate`, I would need to get convinced first by seeing how it's implemented 😄 it's convenient now to not care about the lifecycle, viewmodel gets initialized and that's it from our side, if we want to have `onCreate` then there's some logic requred on composable side to distinguish when it is "first" onCreate and when it is created after state restoration